packages/passwordless-auth-construct/src/custom-auth/custom_auth_service.ts

```diff
@@ -210,9 +216,49 @@ export class CustomAuthService {
 
     const method = this.validateSignInMethod(signInMethod);
 
-    return this.challengeServiceFactory
+    const verifyResult = this.challengeServiceFactory
       .getService(method)
       .verifyChallenge(event);
+
+    const answerCorrect = (await verifyResult).response.answerCorrect;
+    logger.debug(`Answer is correct: ${answerCorrect ? 'true' : 'false'}`);
+
+    try {
+      const passwordlessConfiguration =
+        event.request.userAttributes[PASSWORDLESS_SIGN_UP_ATTR_NAME] &&
+        JSON.parse(
+          event.request.userAttributes[PASSWORDLESS_SIGN_UP_ATTR_NAME]
+        );
+
+      if (passwordlessConfiguration.allowSignInAttempt) {
+        const attributeName =
+          passwordlessConfiguration.deliveryMedium === 'SMS'
+            ? 'phone_number_verified'
+            : 'email_verified';
+
+        const attributeVerified =
+          event.request.userAttributes[attributeName] === 'true';
+        // Only update verified attribute the first time
+        if (answerCorrect && !attributeVerified) {
+          const cognitoUserService = new CognitoUserService(
+            new CognitoIdentityProviderClient({ region: event.region })
+          );
+          logger.debug(`Updating user attribute to verified: ${attributeName}`);
+          await cognitoUserService.markAsVerifiedAndDeletePasswordlessAttribute(
```

Comment:
request: All dependencies should be defined in this [index](https://github.com/aws-amplify/amplify-backend/blob/feat/passwordless-auth/packages/passwordless-auth-construct/src/custom-auth/index.ts). This will need to become a constructor prop on `CustomAuthService`.